workspaces/liferay-customer-workspace/client-extensions/liferay-customer-etc-spring-boot/src/main/java/com/liferay/customer/JiraRestController.java

```diff
@@ -108,33 +145,151 @@ public ResponseEntity<String> get(@PathVariable("issueKey") String issueKey)
 		path = "/jira/security-vulnerabilities/search"
 	)
 	public ResponseEntity<String> search(
-			@RequestParam(defaultValue = "", required = false) String keywords)
+			@RequestParam(defaultValue = "", required = false) String
+				filterAffectedVersions,
+			@RequestParam(defaultValue = "", required = false) String
+				filterCategories,
+			@RequestParam(defaultValue = "", required = false) String
+				filterClassifications,
+			@RequestParam(defaultValue = "", required = false) String
+				filterFixVersions,
+			@RequestParam(defaultValue = "", required = false) String
+				filterSeverities,
+			@RequestParam(defaultValue = "", required = false) String keywords,
+			@RequestParam(defaultValue = "", required = false) String page,
+			@RequestParam(defaultValue = "", required = false) String pageSize,
+			@RequestParam(defaultValue = "", required = false) String sortOrder)
 		throws Exception {
 
 		try {
-			StringBundler sb = new StringBundler(9);
+			StringBundler sb = new StringBundler(49);
 
 			sb.append("project = '");
 			sb.append(_jiraSecurityVulnerabilityProject);
-			sb.append("' AND 'Publishing Status' = 'Ready for Publishing'");
+			sb.append("' AND ");
+			sb.append(
+				_getJQLCustomField(
+					_jiraSecurityVulnerabilityFieldPublishingStatus));
+			sb.append(" = 'Ready for Publishing'");
 
 			if (_isPartner()) {
-				sb.append(" AND 'Partner Publishing Date' <= now()");
+				sb.append(" AND ");
+				sb.append(
+					_getJQLCustomField(
+						_jiraSecurityVulnerabilityFieldPartnerPublishingDate));
+				sb.append(" <= now()");
 			}
 			else {
-				sb.append(" AND 'Customer Publishing Date' <= now()");
+				sb.append(" AND ");
+				sb.append(
+					_getJQLCustomField(
+						_jiraSecurityVulnerabilityFieldCustomerPublishingDate));
+				sb.append(" <= now()");
+			}
+
+			if (Validator.isNotNull(filterAffectedVersions)) {
+				String[] versions = filterAffectedVersions.split(
```

Comment:
can this come in as a string array in the request params? If not, we should use stringutil.split here

Reply:
Looks like it can so far. Tested it with the same console query. Re-pushed.